includes/event/event-repository.php

```diff
@@ -0,0 +1,315 @@
+		return $this->execute_events_query(
+			array(
+				'post_status'    => 'publish',
+				'post__in'       => $event_ids_user_is_attending,
+				'paged'          => $page,
+				'posts_per_page' => $page_size,
+				'meta_query'     => array(
+					array(
+						'key'     => '_event_end',
+						'value'   => $boundary_start->format( 'Y-m-d H:i:s' ),
+						'compare' => '>=',
+						'type'    => 'DATETIME',
+					),
+					array(
+						'key'     => '_event_end',
+						'value'   => $boundary_end->format( 'Y-m-d H:i:s' ),
+						'compare' => '<',
+						'type'    => 'DATETIME',
+					),
+				),
+				'meta_key'       => '_event_end',
+				'meta_type'      => 'DATETIME',
+				'orderby'        => array( 'meta_value', 'ID' ),
+				'order'          => 'DESC',
+			)
+		);
+		// phpcs:enable
+	}
+
+	public function get_events_created_by_user( int $user_id, int $page = -1, int $page_size = -1 ): Events_Query_Result {
+		$this->assert_pagination_arguments( $page, $page_size );
+		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+		return $this->execute_events_query(
+			array(
+				'post_status'    => array( 'publish', 'draft' ),
+				'author'         => $user_id,
+				'paged'          => $page,
+				'posts_per_page' => $page_size,
+				'meta_key'       => '_event_start',
+				'orderby'        => array( 'meta_value', 'ID' ),
+				'order'          => 'DESC',
+			)
+		);
+		// phpcs:enable
+	}
+
+	/**
+	 * @throws Exception
+	 */
+	protected function get_events_active_between(
+		DateTimeImmutable $boundary_start,
+		DateTimeImmutable $boundary_end,
+		array $filter_by_ids = array(),
+		int $page = -1,
+		int $page_size = -1
+	): Events_Query_Result {
+		if ( $boundary_end < $boundary_start ) {
+			throw new Exception( 'boundary end must not be before boundary start' );
+		}
+
+		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+		$query_args = array(
+			'post_status'    => 'publish',
+			'paged'          => $page,
+			'posts_per_page' => $page_size,
+			'meta_query'     => array(
+				array(
+					'key'     => '_event_start',
+					'value'   => $boundary_end->format( 'Y-m-d H:i:s' ),
+					'compare' => '<',
+					'type'    => 'DATETIME',
+				),
+				array(
+					'key'     => '_event_end',
+					'value'   => $boundary_start->format( 'Y-m-d H:i:s' ),
+					'compare' => '>',
+					'type'    => 'DATETIME',
+				),
+			),
+			'meta_key'       => '_event_start',
+			'meta_type'      => 'DATETIME',
+			'orderby'        => array( 'meta_value', 'ID' ),
+		);
+		// phpcs:enable
+
+		if ( ! empty( $filter_by_ids ) ) {
+			$query_args['post__in'] = $filter_by_ids;
+		}
+
+		return $this->execute_events_query( $query_args );
+	}
+
+	/**
+	 * @throws Exception
+	 */
+	protected function assert_pagination_arguments( int $page, int $page_size ) {
```

Comment:
As this function is specific for this class (it is not implementing a method from the interface), it will be interesting to add the description to this method and to the other methods that are specific for only the current class.